app/uk/gov/hmrc/personaldetailsvalidation/model/PersonalDetails.scala

```diff
@@ -46,16 +59,49 @@ object PersonalDetails {
   }
 }
 
+
 case class PersonalDetailsWithNino(firstName: NonEmptyString,
                                    lastName: NonEmptyString,
                                    dateOfBirth: LocalDate,
                                    nino: Nino) extends PersonalDetails with PersonalDetailsNino {
+
+  override def addGender(gender: NonEmptyString): PersonalDetails =
+    PersonalDetailsWithNinoAndGender(firstName, lastName, dateOfBirth, nino, gender)
+
+  override def maybeNino: Option[Nino] = Some(nino)
+
+  def age: Int = YEARS.between(dateOfBirth, LocalDate.now()).toInt
+
   lazy val toJson: JsObject = Json.obj(
     "firstName" -> firstName,
     "lastName" -> lastName,
     "dateOfBirth" -> dateOfBirth,
     "nino" -> nino
   )
+
+
+}
+
+case class PersonalDetailsWithNinoAndGender(firstName: NonEmptyString,
+                                            lastName: NonEmptyString,
+                                            dateOfBirth: LocalDate,
+                                            nino: Nino,
+                                            gender: NonEmptyString) extends PersonalDetails with PersonalDetailsNino with PersonalDetailsGender {
+
+  override def maybeNino: Option[Nino] = Some(nino)
+
+  override def maybeGender: Option[NonEmptyString] = Some(gender)
```

Comment:
Again, the class name would suggest that Gender is always part of this data model?